routes/user.js

```diff
@@ -9,48 +10,80 @@ var router = express.Router();
 var log = require("../logger");
 
 router.route("/")
-    .all(permissions.requireAll({type: "USER_MANAGE"}))
+// .all(permissions.requireAll({type: "USER_MANAGE"}))
     .get(function (req, res, next) {
-        User.findAll({
-          attributes: ["id", "displayName", "email", "isAdmin"],
-          order: [
-            ["id", "ASC"]
-          ]
-        }).then(function (results) {
-            res.send(results);
-        });
+        var userId = res.locals.session ? res.locals.session.user : null;
+        permissions.check(userId, {
+            type: "USER_MANAGE",
+            value: userId
+        }).then(function (result) {
+            if (!result) res.sendStatus(403);
+            User.findAll({
+                attributes: ["id", "displayName", "email", "isAdmin"],
+                order: [
+                    ["id", "ASC"]
+                ]
+            }).then(function (results) {
+                res.send(results);
+            });
+        }).done();
     })
     .post(function (req, res, next) {
-        log.info({req: req.body}, "INITIAL REQUEST");
-        if (!res.locals.session) {
-            return res.sendStatus(401);
-        }
+        var makeLink = function (length) {
+            var result = '';
+            var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+            var charactersLength = characters.length;
```

Comment:
seems pointless to define this here, instead of just using characters.length in the selection function

Reply:
efficiency to not recalculate it again?